MyPy `[return-value]` error in `InlineKeyboardBuilder().as_markup()`

## MyPy marking code below as error.

```shell
language.py: note: In function "get_lang_keyboard":
language.py:76:9: error: Incompatible return value type (got
"InlineKeyboardMarkup | ReplyKeyboardMarkup", expected "InlineKeyboardMarkup")  [return-value]                                                        
            InlineKeyboardBuilder()
            ^
```

```python
def get_lang_keyboard(user_id: int, i18n: I18nContext) -> InlineKeyboardMarkup:
    return (
        InlineKeyboardBuilder()
        .row(
            *[
                InlineKeyboardButton(
                    text=i18n.settings.language.code(language_code=language.value, _path="cmds/user_settings.ftl"),
                    callback_data=LanguageCallbackData(owner_id=user_id, language=language).pack(),
                )
                for language in PossibleLanguages
            ],
            width=2,
        )
        .as_markup()
    )
```

--- a/CHANGES/1677.misc.rst
+++ b/CHANGES/1677.misc.rst
@@ -1,1 +1,5 @@
-Fixed an issue #1677
+Fixed MyPy [return-value] error in `InlineKeyboardBuilder().as_markup()`.
+`as_markup` method now overloads parent class method and uses `super()`, to call parent's 
+`as_markup` method.
+Also added correct type hint to `as_markup`'s return in `InlineKeyboardBuilder` and 
+`ReplyKeyboardBuilder` classes.

--- a/CHANGES/1677.misc.rst
+++ b/CHANGES/1677.misc.rst
@@ -1,0 +1,1 @@
+Fixed an issue #1677

--- a/aiogram/utils/keyboard.py
+++ b/aiogram/utils/keyboard.py
@@ -336,11 +336,9 @@ class InlineKeyboardBuilder(KeyboardBuilder[InlineKeyboardButton]):
             ),
         )
 
-    if TYPE_CHECKING:
-
-        def as_markup(self, **kwargs: Any) -> InlineKeyboardMarkup:
-            """Construct an InlineKeyboardMarkup"""
-            ...
+    def as_markup(self, **kwargs: Any) -> InlineKeyboardMarkup:
+        """Construct an InlineKeyboardMarkup"""
+        return cast(InlineKeyboardMarkup, super().as_markup(**kwargs))
 
     def __init__(self, markup: Optional[List[List[InlineKeyboardButton]]] = None) -> None:
         super().__init__(button_type=InlineKeyboardButton, markup=markup)
@@ -401,9 +399,9 @@ class ReplyKeyboardBuilder(KeyboardBuilder[KeyboardButton]):
             ),
         )
 
-    if TYPE_CHECKING:
-
-        def as_markup(self, **kwargs: Any) -> ReplyKeyboardMarkup: ...
+    def as_markup(self, **kwargs: Any) -> ReplyKeyboardMarkup:
+        """Construct a ReplyKeyboardMarkup"""
+        return cast(ReplyKeyboardMarkup, super().as_markup(**kwargs))
 
     def __init__(self, markup: Optional[List[List[KeyboardButton]]] = None) -> None:
         super().__init__(button_type=KeyboardButton, markup=markup)

